### PR TITLE
add func (a Amount) Scaled() int64

### PR DESCRIPTION
**Pull request by mwuertinger**
Usecases:
- Custom serialization / deserialization
- Persisting Amount as two columns in a database

### CODE REVIEW

File: amount.go [hunk 1]
```diff
@@ -35,6 +35,9 @@ func (a Amount) Float64() float64 { return float64(a.v) / float64(a.c.scale()) }
 
 func (a Amount) Currency() Currency { return a.c }
 
+// Val returns the raw integer value.
+func (a Amount) Val() int64 { return a.v }
+
```
Comment (nikolaydubina) [inline amount.go]: lets use same notation as in fpdecimal. https://github.com/nikolaydubina/fpdecimal/blob/main/fpdecimal.go#L45
```suggestion
func (a Amount) Scaled() int64 { return a.v }

```

**Review by nikolaydubina — CHANGES_REQUESTED** [review]
sure, makes sense.

please add test for this though illustrating that this value is multiplied by 100. perhaps Example test is very nice here.

**Comment by mwuertinger** [comment]
Good idea! Added a test.

File: amount_test.go [hunk 2]
```diff
@@ -76,6 +76,26 @@ func ExampleFromFloat() {
...
+		{fpmoney.FromFloat(8764534896.42, fpmoney.USD), 876453489642},
+	}
+
+	for _, d := range data {
+		res := d.a.Scaled()
+		if res != d.expected {
+			t.Errorf("%v: expected: %d, got: %d", d, d.expected, res)
+		}
+	}
+}
```
Comment (nikolaydubina) [inline amount_test.go]: ah, let's make it an Example. this is something people should know when they reading documentation

```suggestion
func ExampleAmount_Scaled_fractions() {
	v := fpmoney.FromFloat(42.23, fpmoney.EUR)
	fmt.Println(v.Scaled())
	// Output: 4223
}

func ExampleAmount_Scaled_many_fractions() {
	v := fpmoney.FromFloat(17.0, fpmoney.CLF)
	fmt.Println(v.Scaled())
	// Output: 170000
}

func ExampleAmount_Scaled_large() {
	v := fpmoney.FromFloat(8764534896.42, fpmoney.USD)
	fmt.Println(v.Scaled())
	// Output: 876453489642
}
func ExampleAmount_Scaled_whole() {
	v := fpmoney.FromFloat(23.0, fpmoney.EUR)
	fmt.Println(v.Scaled())
	// Output: 2300
}

func ExampleAmount_Scaled_from_scaled() {
	v := fpmoney.FromIntScaled(17, fpmoney.EUR)
	fmt.Println(v.Scaled())
	// Output: 17
}
```

**Review by nikolaydubina — APPROVED** [review]
Screenshot 2024-09-03 at 21 49 37

**Pull request merged**